src/Embeddings/VectorStores/Elasticsearch/ElasticsearchVectorStore.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+namespace LLPhant\Embeddings\VectorStores\Elasticsearch;
+
+use Elastic\Elasticsearch\Client;
+use Elastic\Elasticsearch\Exception\ClientResponseException;
+use Elastic\Elasticsearch\Exception\MissingParameterException;
+use Elastic\Elasticsearch\Exception\ServerResponseException;
+use LLPhant\Embeddings\Document;
+use LLPhant\Embeddings\VectorStores\VectorStoreBase;
+
+class ElasticsearchVectorStore extends VectorStoreBase
+{
+    final public const INDEX_NAME = 'llphant';
+
+    /**
+     * @throws \Throwable
+     * @throws ClientResponseException
+     * @throws ServerResponseException
+     * @throws MissingParameterException
+     */
+    public function __construct(public Client $client, public int $vectorDim, public string $indexName = self::INDEX_NAME)
```

Comment:
Ici on est obligé d'entrer la dimension des vecteurs. On pourrait rendre l'option facultative et déterminer la dimension lors de la soumission du 1er vecteur (dans addDocument/addDocuments) et mettre à jour le schema de l'index avec la bonne dimension. T'en dis quoi @MaximeThoonsen ?

Reply:
yes it could be nice.